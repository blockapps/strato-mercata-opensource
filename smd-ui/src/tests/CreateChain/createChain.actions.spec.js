import {
  openCreateChainOverlay,
  closeCreateChainOverlay,
  openAddMemberModal,
  closeAddMemberModal,
  createChain,
  createChainSuccess,
  createChainFailure,
  resetError,
  compileChainContract,
  compileChainContractSuccess,
  compileChainContractFailure,
  resetContract
} from "../../components/CreateChain/createChain.actions";
import { xabiMock } from "./createChainMock";

describe('CreateChain: action', () => {

  test('open overlay', () => {
    expect(openCreateChainOverlay()).toMatchSnapshot();
  });

  test('close overlay', () => {
    expect(closeCreateChainOverlay()).toMatchSnapshot();
  });

  describe('Add Member Modal', () => {

    test('open overlay', () => {
      expect(openAddMemberModal()).toMatchSnapshot();
    });

    test('close overlay', () => {
      expect(closeAddMemberModal()).toMatchSnapshot();
    });

  })

  describe('create chain', () => {

    const payload = {
      label: 'airline cartel 9',
      members: [{ address: "f11b5c42f5b84efa07f6b0a32c3fc545ff509126", enode: "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7â€¦4cac9f77166ad92a0@172.16.0.5:30303?discport=30303" }],
      balances: [{ balance: 500000000000000, address: "f11b5c42f5b84efa07f6b0a32c3fc545ff509126" }],
      src: `contract SimpleStorage {
        uint public storedData;
      }`,
      args: { addRule: "MajorityRules", removeRule: "MajorityRules" },
      contractName: 'SimpleStorage',
      vm: false,
      limit: 10,
      offset: 0
    }

    test('request', () => {
      expect(createChain(payload.label, payload.members, payload.balances, payload.src, payload.args, payload.vm, payload.contractName, payload.limit, payload.offset)).toMatchSnapshot();
    });

    test('success', () => {
      expect(createChainSuccess('64885c49cdc6fe5f15975596115a120ec1e9a616e88a22e0be0457f373d75b73')).toMatchSnapshot();
    });

    test('failure', () => {
      expect(createChainFailure('error')).toMatchSnapshot();
    });

  });

  describe('compile chain contract', () => {

    const payload = {
      name: "Governance",
      contract: "contract Governance { string constant addRule = 'MajorityRules'; string constant removeRule = 'MajorityRules' }",
      searchable: false,
      vm: false,
    }

    test('request', () => {
      expect(compileChainContract(payload.label, payload.contract, payload.searchable, payload.vm)).toMatchSnapshot();
    });

    test('success', () => {
      expect(compileChainContractSuccess(xabiMock)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(compileChainContractFailure('error')).toMatchSnapshot();
    });

  });

  test('reset error', () => {
    expect(resetError()).toMatchSnapshot();
  });

  test('reset contract', () => {
    expect(resetContract()).toMatchSnapshot();
  });

});