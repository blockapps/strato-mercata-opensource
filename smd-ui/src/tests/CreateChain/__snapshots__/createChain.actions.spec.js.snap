// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateChain: action Add Member Modal close overlay 1`] = `
Object {
  "isOpen": false,
  "type": "CLOSE_ADD_MEMBER_MODAL",
}
`;

exports[`CreateChain: action Add Member Modal open overlay 1`] = `
Object {
  "isOpen": true,
  "type": "OPEN_ADD_MEMBER_MODAL",
}
`;

exports[`CreateChain: action close overlay 1`] = `
Object {
  "isOpen": false,
  "type": "CLOSE_CHAIN_MODAL",
}
`;

exports[`CreateChain: action compile chain contract failure 1`] = `
Object {
  "error": "error",
  "type": "COMPILE_CHAIN_CONTRACT_FAILURE",
}
`;

exports[`CreateChain: action compile chain contract request 1`] = `
Object {
  "contract": "contract Governance { string constant addRule = 'MajorityRules'; string constant removeRule = 'MajorityRules' }",
  "name": undefined,
  "searchable": false,
  "type": "COMPILE_CHAIN_CONTRACT_REQUEST",
  "vm": false,
}
`;

exports[`CreateChain: action compile chain contract success 1`] = `
Object {
  "response": Object {
    "src": Object {
      "Governance": Object {
        "constr": Object {},
        "events": Object {},
        "funcs": Object {},
        "isLibrary": false,
        "modifiers": Object {},
        "types": Object {},
        "using": Object {},
        "vars": Object {
          "addRule": Object {
            "atBytes": 0,
            "constant": true,
            "dynamic": true,
            "initialValue": "\\"MajorityRules\\"",
            "public": null,
            "type": "String",
          },
          "removeRule": Object {
            "atBytes": 0,
            "constant": true,
            "dynamic": true,
            "initialValue": "\\"MajorityRules\\"",
            "public": null,
            "type": "String",
          },
          "temp": Object {
            "atBytes": 0,
            "constant": null,
            "dynamic": true,
            "initialValue": null,
            "public": null,
            "type": "String",
          },
        },
      },
    },
  },
  "type": "COMPILE_CHAIN_CONTRACT_SUCCESS",
}
`;

exports[`CreateChain: action create chain failure 1`] = `
Object {
  "error": "error",
  "isOpen": false,
  "spinning": false,
  "type": "CREATE_CHAIN_FAILURE",
}
`;

exports[`CreateChain: action create chain request 1`] = `
Object {
  "args": false,
  "balances": Array [
    Object {
      "address": "f11b5c42f5b84efa07f6b0a32c3fc545ff509126",
      "balance": 500000000000000,
    },
  ],
  "contractName": 10,
  "integrations": "contract SimpleStorage {
        uint public storedData;
      }",
  "isOpen": true,
  "label": "airline cartel 9",
  "limit": 0,
  "members": Array [
    Object {
      "address": "f11b5c42f5b84efa07f6b0a32c3fc545ff509126",
      "enode": "enode://6f8a80d14311c39f35f516fa664deaaaa13e85b2f7â€¦4cac9f77166ad92a0@172.16.0.5:30303?discport=30303",
    },
  ],
  "offset": undefined,
  "spinning": true,
  "src": Object {
    "addRule": "MajorityRules",
    "removeRule": "MajorityRules",
  },
  "type": "CREATE_CHAIN_REQUEST",
  "vm": "SimpleStorage",
}
`;

exports[`CreateChain: action create chain success 1`] = `
Object {
  "isOpen": false,
  "key": "64885c49cdc6fe5f15975596115a120ec1e9a616e88a22e0be0457f373d75b73",
  "spinning": false,
  "type": "CREATE_CHAIN_SUCCESS",
}
`;

exports[`CreateChain: action open overlay 1`] = `
Object {
  "isOpen": true,
  "type": "OPEN_CHAIN_MODAL",
}
`;

exports[`CreateChain: action reset contract 1`] = `
Object {
  "type": "RESET_CONTRACT",
}
`;

exports[`CreateChain: action reset error 1`] = `
Object {
  "type": "RESET_ERROR",
}
`;
